scintillometry/dispersion/tests/test_dispersion.py

```diff
@@ -0,0 +1,65 @@
+# Licensed under the GPLv3 - see LICENSE
+import pytest
+import numpy as np
+import astropy.units as u
+from astropy.tests.helper import assert_quantity_allclose
+
+from ..dm import DispersionMeasure
+
+
+class TestDM(object):
+
+    def setup(self):
+        self.dm_val = 29.1168    # DM of B1957+20.
+
+    def test_quantity(self):
+        """Test `~astropy.unit.Quantity` creation and unit conversion."""
+        dm = DispersionMeasure(self.dm_val)
+        assert dm == self.dm_val * u.pc / u.cm**3
+        dm_explicitunit = DispersionMeasure(self.dm_val, unit=dm.unit)
+        assert dm == dm_explicitunit
+        newunit = u.lyr / u.m**3
+        dm_differentunit = DispersionMeasure(dm.to_value(newunit),
+                                             unit=newunit)
+        assert dm_differentunit == dm
+        with pytest.raises(u.UnitsError) as excinfo:
+            DispersionMeasure(self.dm_val * u.s)
+        assert "dispersion measures should have units" in str(excinfo.value)
+        # Check that we can copy (and the new object is the same class).
+        dm_copy = dm.copy()
+        assert dm_copy == dm.copy()
+        assert dm_copy.__class__ == dm.__class__
+
+    def test_calculation(self):
+        """Test time and phase offset calculation."""
+        # Array of random frequencies within 64 MHz of the reference frequency.
```

Comment:
This is fine for one test, but I think we should also have a test with the correct number. I.e., use dm=1, f=1MHz and check that you get a time delay of `4148.808 s`